cpp/src/binaryop/binaryop.cpp

```diff
@@ -349,23 +349,26 @@ std::unique_ptr<column> make_fixed_width_column_for_output(column_view const& lh
  */
 bool is_basic_arithmetic_binop(binary_operator op)
 {
-  return op == binary_operator::ADD or  ///< operator +
-         op == binary_operator::SUB or  ///< operator -
-         op == binary_operator::MUL or  ///< operator *
-         op == binary_operator::DIV;    ///< operator / using common type of lhs and rhs
+  return op == binary_operator::ADD or       ///< operator +
+         op == binary_operator::SUB or       ///< operator -
+         op == binary_operator::MUL or       ///< operator *
+         op == binary_operator::DIV or       ///< operator / using common type of lhs and rhs
+         op == binary_operator::NULL_MIN or  ///< 2 null = null, 1 null = value, else min
+         op == binary_operator::NULL_MAX;    ///< 2 null = null, 1 null = value, else max
```

Comment:
Having `min` and `max` be arithmetic ops seems counter-intuitive.

Reply:
Really? Aren't they arithmetic?

Reply:
Or is it because of the `NULL` component.

Reply:
I see, strings are supported. :joy: at having convo with myself

Reply:
Thinking too much in APL, in APL `MAX` doesn't work on strings:
```dyalog
      'acb'⌈'cat'
DOMAIN ERROR
```
I've modified the code.

Reply:
It seemed like the original intention was to distinguish between arithmetic and comparison operations. I'm fine defining min/max as arithmetic, but then why is only `NULL_MIN/NULL_MAX` included and not normal `MIN/MAX`? I'm just asking for consistency.

Reply:
@jrhemstad There is no `MIN` or `MAX`, unless I'm looking in the wrong place

Reply:
**Jake Hemstad  1:36 PM**
that's v surprising
...
**Conor Hoekstra  1:40 PM**
comparison ops are predicates, they return booleans
```
+ - ÷ × ⌈ ⌊   ⍝ all math ops (return number)
< ≤ = ≥ > ∨ ∧ ⍝ all comparison or logical ops (return boolean)
```
**Jake Hemstad  1:43 PM**
sure, that makes sense. I just get itchy when a single function starts making fundamental definitions like this instead of it being defined more globally/programmatically